### ISSUE TITLE
TabBarSelectorBehavior not working correctly for WASM

### ISSUE THREAD

**Issue by agneszitte-nventive** [unoplatform/uno.toolkit.ui#57]
## Current behavior

TabBarSelectorBehavior not working correctly for WASM

## Expected behavior

TabBarSelectorBehavior should work correctly for WASM

## How to reproduce it (as minimally and precisely as possible)

Test the TabBarSelectorBehavior sample
Switch between tabs and notices the issues
![15770841d5caf436cf1b590eac5d6595](https://user-images.githubusercontent.com/16295702/131928963-604263be-4371-4e56-a80c-f640d37c1eb0.gif)

## Environment

Nuget Package: Uno.UI.Toolkit

Package Version(s): 0.1.0-dev.15

Affected platform(s):

- [ ] iOS
- [ ] Android
- [X] WebAssembly
- [ ] WebAssembly renders for Xamarin.Forms
- [ ] Windows
- [ ] Build tasks

Visual Studio:

- [ ] 2017 (version: )
- [ ] 2019 (version: )
- [ ] 2022 (version: 7.0.0 Preview 2.1)
- [ ] for Mac (version: )

**Comment by davidjohnoliver** [comment]
Related issue: https://github.com/unoplatform/uno/issues/7671

**Comment by Xiaoy312** [comment]
<details>
<summary>debug note</summary>

```

# Chapter 0: Uno.Toolkit

@ samples\Uno.Toolkit.Samples\Uno.Toolkit.Samples.Shared\Content\Controls\TabBarBehaviorSamplePage.xaml
    <utu:TabBar utu:TabBarSelectorBehavior.Selector="{Binding ElementName=SlideFlipView}"
                Style="{StaticResource TabBarWithSlideIndicatorStyle}">
        ^ no code-behind, just pure xaml

    layout:
        utu:TabBar.Items > utu:TabBarItem[Content='Tab $'] * 3
        FlipView.Items > (FlipViewItem > Grid[Width=Height=100, Background=$]) * {blue,red,green}}

skia: freezes when selecting a tab...

key events:
    -- loading --
    0 @xy OnSelectorSelectionChanged: index=0, _isSynchronizing=False
    1 @xy OnTabBarSelectionChanged: index=0, _isSynchronizing=False
    -- tapping on tab 2 --
    2 @xy OnTabBarSelectionChanged: index=1, _isSynchronizing=False
    3 @xy OnSelectorSelectionChanged: index=1, _isSynchronizing=True
    4 @xy OnSelectorSelectionChanged: index=0, _isSynchronizing=False
    5 @xy OnTabBarSelectionChanged: index=0, _isSynchronizing=True

    ^ between 3 & 4, there is like ~20 log messages in format of $"ViewChanged: offset={x}" with x going from 1615 to 813

call stack of reset event (log#3)
    full stack:
        at Uno.Toolkit.UI.TabBarSelectorBehaviorState.OnSelectorSelectionChanged(Object sender, SelectionChangedEventArgs e) in D:\code\uno\platform\Uno.UI.Toolkit\src\Uno.Toolkit.UI\Behaviors\TabBarSelectorBehaviorState.cs:line 217
        at Windows.UI.Xaml.Controls.Primitives.Selector.InvokeSelectionChanged(Object[] removedItems, Object[] addedItems)
        at Windows.UI.Xaml.Controls.Primitives.Selector.OnSelectedItemChanged(Object oldSelectedItem, Object selectedItem, Boolean updateItemSelectedState)
        at Windows.UI.Xaml.Controls.Primitives.Selector.OnPropertyChanged2(DependencyPropertyChangedEventArgs args)
        at Windows.UI.Xaml.DependencyObjectStore.InvokeCallbacks(DependencyObject actualInstanceAlias, DependencyProperty property, DependencyPropertyDetails propertyDetails, Object previousValue, DependencyPropertyValuePrecedences previousPrecedence, Object newValue, DependencyPropertyValuePrecedences newPrecedence, Boolean bypassesPropagation)
        at Windows.UI.Xaml.DependencyObjectStore.RaiseCallbacks(DependencyObject actualInstanceAlias, DependencyPropertyDetails propertyDetails, Object previousValue, DependencyPropertyValuePrecedences previousPrecedence, Object newValue, DependencyPropertyValuePrecedences newPrecedence)
        at Windows.UI.Xaml.DependencyObjectStore.InnerSetValue(DependencyProperty property, Object value, DependencyPropertyValuePrecedences precedence, DependencyPropertyDetails propertyDetails, Boolean isPersistentResourceBinding)
        at Windows.UI.Xaml.DependencyObjectStore.SetValue(DependencyProperty property, Object value, DependencyPropertyValuePrecedences precedence, DependencyPropertyDetails propertyDetails, Boolean isPersistentResourceBinding)
        at Windows.UI.Xaml.DependencyObjectStore.SetValue(DependencyProperty property, Object value)
        at Windows.UI.Xaml.UIElement.SetValue(DependencyProperty dp, Object value)
        at Windows.UI.Xaml.Controls.Primitives.Selector.set_SelectedItem(Object value)
        at Windows.UI.Xaml.Controls.Primitives.Selector.OnSelectedIndexChanged(Int32 oldSelectedIndex, Int32 newSelectedIndex)
        at Windows.UI.Xaml.Controls.FlipView.OnSelectedIndexChanged(Int32 oldSelectedIndex, Int32 newSelectedIndex)
        at Windows.UI.Xaml.Controls.Primitives.Selector.OnPropertyChanged2(DependencyPropertyChangedEventArgs args)
        at Windows.UI.Xaml.DependencyObjectStore.InvokeCallbacks(DependencyObject actualInstanceAlias, DependencyProperty property, DependencyPropertyDetails propertyDetails, Object previousValue, DependencyPropertyValuePrecedences previousPrecedence, Object newValue, DependencyPropertyValuePrecedences newPrecedence, Boolean bypassesPropagation)
        at Windows.UI.Xaml.DependencyObjectStore.RaiseCallbacks(DependencyObject actualInstanceAlias, DependencyPropertyDetails propertyDetails, Object previousValue, DependencyPropertyValuePrecedences previousPrecedence, Object newValue, DependencyPropertyValuePrecedences newPrecedence)
        at Windows.UI.Xaml.DependencyObjectStore.InnerSetValue(DependencyProperty property, Object value, DependencyPropertyValuePrecedences precedence, DependencyPropertyDetails propertyDetails, Boolean isPersistentResourceBinding)
        at Windows.UI.Xaml.DependencyObjectStore.SetValue(DependencyProperty property, Object value, DependencyPropertyValuePrecedences precedence, DependencyPropertyDetails propertyDetails, Boolean isPersistentResourceBinding)
        at Windows.UI.Xaml.DependencyObjectStore.SetValue(DependencyProperty property, Object value)
        at Windows.UI.Xaml.UIElement.SetValue(DependencyProperty dp, Object value)
        at Windows.UI.Xaml.Controls.Primitives.Selector.set_SelectedIndex(Int32 value)
        at Windows.UI.Xaml.Controls.FlipView.OnScrollViewerViewChanged(Object pSender, ScrollViewerViewChangedEventArgs pArgs)
        at Windows.UI.Xaml.Controls.ScrollViewer.Update(Boolean isIntermediate)
        at Windows.UI.Xaml.Controls.ScrollViewer.OnPresenterScrolled(Double horizontalOffset, Double verticalOffset, Boolean isIntermediate)
        at Windows.UI.Xaml.Controls.ScrollContentPresenter.OnScroll(Object sender, EventArgs args)
        at Windows.UI.Xaml.UIElement.GenericEventHandlers.RaiseEventHandler(Delegate d, Object sender, Object args)
        at Windows.UI.Xaml.UIElement.EventRegistration.Dispatch(EventArgs eventArgs, String nativeEventPayload)
        at Windows.UI.Xaml.UIElement.InternalDispatchEvent(String eventName, EventArgs eventArgs, String nativeEventPayload)
        at Windows.UI.Xaml.UIElement.DispatchEvent(Int32 handle, String eventName, String eventArgs)
    trimmed:
        at Uno.Toolkit.UI.TabBarSelectorBehaviorState.OnSelectorSelectionChanged(Object sender, SelectionChangedEventArgs e) in D:\code\uno\platform\Uno.UI.Toolkit\src\Uno.Toolkit.UI\Behaviors\TabBarSelectorBehaviorState.cs:line 217
        at Windows.UI.Xaml.Controls.Primitives.Selector.InvokeSelectionChanged(Object[] removedItems, Object[] addedItems)
        at Windows.UI.Xaml.Controls.Primitives.Selector.OnSelectedItemChanged(Object oldSelectedItem, Object selectedItem, Boolean updateItemSelectedState)
        at Windows.UI.Xaml.Controls.Primitives.Selector.OnPropertyChanged2(DependencyPropertyChangedEventArgs args)
        at Windows.UI.Xaml.Controls.Primitives.Selector.set_SelectedItem(Object value)
        at Windows.UI.Xaml.Controls.Primitives.Selector.OnSelectedIndexChanged(Int32 oldSelectedIndex, Int32 newSelectedIndex)
        
        at Windows.UI.Xaml.Controls.FlipView.OnSelectedIndexChanged(Int32 oldSelectedIndex, Int32 newSelectedIndex)
        at Windows.UI.Xaml.Controls.Primitives.Selector.OnPropertyChanged2(DependencyPropertyChangedEventArgs args)
        at Windows.UI.Xaml.Controls.Primitives.Selector.set_SelectedIndex(Int32 value)

        at Windows.UI.Xaml.Controls.FlipView.OnScrollViewerViewChanged(Object pSender, ScrollViewerViewChangedEventArgs pArgs)
        at Windows.UI.Xaml.Controls.ScrollViewer.Update(Boolean isIntermediate)
        at Windows.UI.Xaml.Controls.ScrollViewer.OnPresenterScrolled(Double horizontalOffset, Double verticalOffset, Boolean isIntermediate)
        at Windows.UI.Xaml.Controls.ScrollContentPresenter.OnScroll(Object sender, EventArgs args)

FlipView itself just doesnt work...
    always stuck on the first blue square, you can try to navigate to the next, 
    but the content always reset/rewind back
    also the previous button never appears...
    when navigating, the content instantly becomes blank and the blue square pan in from the left
    as if, we jump instantly to item#-1 and then animated toward item#0

# Chapter 1: Uno.UI
@ src\SamplesApp\UITests.Shared\Windows_UI_Xaml_Controls\FlipView\FlipView_Buttons.xaml
    <FlipView Width="200" Height="100" x:Name="SUT">
        <Button x:Name="Button1" Content="Button 1" Width="200" Background="Green"  AutomationProperties.Name="Button1" />
        ...
        <Button x:Name="Button5" Content="Button 5" Width="200" Background="Orange" AutomationProperties.Name="Button5" />

default template:
    D:\code\uno\framework\uno\src\Uno.UI\UI\Xaml\Style\Generic\Generic.xaml @L7713

flow:
    --- pressing next button
    @xy ===== MoveNext()
    @xy ===== UpdateSelectedIndex(index: 1)
    @xy ===== OnSelectedIndexChanged(oldSelectedIndex: 0, newSelectedIndex: 1)
    @xy { oldIndex = 0, newIndex = 1, shouldUseAnimation = True, m_skipAnimationOnce = False }
    @xy { m_animateNewIndex = False }
    @xy if true
    ... animating
    @xy ===== OnScrollViewerViewChanged(pSender: Windows.UI.Xaml.Controls.ScrollViewer, pArgs.IsIntermediate: False)
    @xy { offset = 196, viewportSize = 196, m_animateNewIndex = True }
    @xy ===== OnScrollViewerViewChanged(pSender: Windows.UI.Xaml.Controls.ScrollViewer, pArgs.IsIntermediate: False)
    @xy { offset = 196, viewportSize = 196, m_animateNewIndex = False }
    @xy nSelectedIndex = 1
    @xy ===== OnScrollViewerViewChanged(pSender: Windows.UI.Xaml.Controls.ScrollViewer, pArgs.IsIntermediate: False)
    @xy { offset = 193, viewportSize = 196, m_animateNewIndex = False }
    @xy nSelectedIndex = 1
    ... (skipping multiple lines... offset 193->121)
    @xy ===== OnScrollViewerViewChanged(pSender: Windows.UI.Xaml.Controls.ScrollViewer, pArgs.IsIntermediate: False)
    @xy { offset = 121, viewportSize = 196, m_animateNewIndex = False }
    @xy nSelectedIndex = 1
    @xy ===== OnScrollViewerViewChanged(pSender: Windows.UI.Xaml.Controls.ScrollViewer, pArgs.IsIntermediate: False)
    @xy { offset = 98, viewportSize = 196, m_animateNewIndex = False }
    @xy nSelectedIndex = 0
    ... resetting
    @xy ===== OnSelectedIndexChanged(oldSelectedIndex: 1, newSelectedIndex: 0)
    @xy { oldIndex = 1, newIndex = 0, shouldUseAnimation = True, m_skipAnimationOnce = True }
    @xy { m_animateNewIndex = False }
    @xy else
    @xy ===== SetOffsetToSelectedIndex()
    @xy ===== OnScrollViewerViewChanged(pSender: Windows.UI.Xaml.Controls.ScrollViewer, pArgs.IsIntermediate: False)
    @xy { offset = 96, viewportSize = 196, m_animateNewIndex = False }
    @xy nSelectedIndex = 0

original source: UNDER_NDA: dxaml\xcp\dxaml\lib\FlipView_Partial.cpp

@ MoveNext -> UpdateSelectedIndex -> OnSelectedIndexChanged (main branch): -> 
    m_animateNewIndex = true;
    m_tpScrollViewer?.BringIntoViewport(...);
        "-> OnScrollViewerViewChanged\n" * n // from scrolling via (js:)scrollTo

@ OnScrollViewerViewChanged
    /* !!! branching inverted for simplicity */
    if (m_animateNewIndex) m_animateNewIndex = false;
    else
    { 
        /* GIVEN there is no pending changes */
        m_skipAnimationOnce = true;
        SelectedIndex = (int)(Math.Round(bIsVirtualizing ? ItemsPresenter.OffsetToIndex(offset) : (offset / viewportSize), 0));
        m_skipAnimationOnce = false;
    }

    ^ without touching `m_animateNewIndex`, if the else-branch (inverted) were always execute, the scroll is fine in such case

## Clipping
^ however 2nd item and beyond are not rendering, despite being scrolled there
    ^ are we perhaps not rendering/materializing them?
    ^ looking at the html, they are materialized, the frames are correct, and everything..? wtf

^ the items after the 1st got clipped by `THIS_PANEL_HERE`
    FlipView > Grid > ScrollViewer > Border > Grid > ScrollContentPresenter > ItemsPresenter > StackPanel.THIS_PANEL_HERE > FlipViewItem * 5
    <div xamltype="...StackPanel" style="...;clip: rect(0px, 200px, 100px, 0px);">
        ^ the width (2nd arg) needs to be 1000px (200px * 5 items)

@xy StackPanel#278034
    @xy { customClipping = True, allow = True, forced = False }
    @xy [Windows.UI.Xaml.Controls.StackPanel/: InnerArrangeCore(0,0,200,100)
    @xy [Windows.UI.Xaml.Controls.StackPanel/: InnerArrangeCore(0,0,200,100) - allowClip=True, arrangeSize=200,100, _unclippedDesiredSize=1000,100, forcedClipping=False
    @xy [Windows.UI.Xaml.Controls.StackPanel/: (arrangeSize.Width) 200 < 1000: NEEDS CLIPPING.
    @xy [Windows.UI.Xaml.Controls.StackPanel/: InnerArrangeCore(0,0,200,100) - effectiveMaxSize=Infinity,Infinity, maxSize=Infinity,Infinity, _unclippedDesiredSize=1000,100, forcedClipping=True
    @xy [Windows.UI.Xaml.Controls.StackPanel/: ArrangeOverride(200,100)=200,100, clipped=200,100 (max=Infinity,Infinity) needsClipToSlot=True
    @xy [[Windows.UI.Xaml.Controls.StackPanel/] ArrangeChild(offset=[0, 0], margin=[Thickness: 0-0-0-0]) [oldRenderSize=0,0] [RenderSize=200,100] [clippedInkSize=200,100] [RequiresClipping=True]

@ src\Uno.UI\UI\Xaml\Controls\Panel\Panel.wasm.cs
    bool ICustomClippingElement.AllowClippingToLayoutSlot => true;
    ^ changing that to false prevents the clipping
    ^ all other platforms has this as false, but this was true... WHY?

## Manipulation
fail: Windows.UI.Xaml.UIElement[0]
    Failed to raise 'PointerReleasedEvent': System.NotImplementedException: The member bool UIElement.CancelDirectManipulations() is not implemented in Uno.
        at Windows.UI.Xaml.UIElement.CancelDirectManipulations () <0xfdc0878 + 0x0000e> in <filename unknown>:0 
        at Windows.UI.Xaml.Controls.FlipView.OnSelectedIndexChanged (System.Int32 oldSelectedIndex, System.Int32 newSelectedIndex) <0xff189a8 + 0x0034a> in <filename unknown>:0 
        at Windows.UI.Xaml.Controls.Primitives.Selector.OnPropertyChanged2 (Windows.UI.Xaml.DependencyPropertyChangedEventArgs args) <0xfca7758 + 0x000c0> in <filename unknown>:0 
        at Windows.UI.Xaml.DependencyObjectStore.InvokeCallbacks (Windows.UI.Xaml.DependencyObject actualInstanceAlias, Windows.UI.Xaml.DependencyProperty property, Windows.UI.Xaml.DependencyPropertyDetails propertyDetails, System.Object previousValue, Windows.UI.Xaml.DependencyPropertyValuePrecedences previousPrecedence, System.Object newValue, Windows.UI.Xaml.DependencyPropertyValuePrecedences newPrecedence, System.Boolean bypassesPropagation) <0xde04298 + 0x00252> in <filename unknown>:0 
        at Windows.UI.Xaml.DependencyObjectStore.RaiseCallbacks (Windows.UI.Xaml.DependencyObject actualInstanceAlias, Windows.UI.Xaml.DependencyPropertyDetails propertyDetails, System.Object previousValue, Windows.UI.Xaml.DependencyPropertyValuePrecedences previousPrecedence, System.Object newValue, Windows.UI.Xaml.DependencyPropertyValuePrecedences newPrecedence) <0xde031a0 + 0x00100> in <filename unknown>:0 
        at Windows.UI.Xaml.DependencyObjectStore.InnerSetValue (Windows.UI.Xaml.DependencyProperty property, System.Object value, Windows.UI.Xaml.DependencyPropertyValuePrecedences precedence, Windows.UI.Xaml.DependencyPropertyDetails propertyDetails, System.Boolean isPersistentResourceBinding) <0xddebc18 + 0x00386> in <filename unknown>:0 
        at Windows.UI.Xaml.DependencyObjectStore.SetValue (Windows.UI.Xaml.DependencyProperty property, System.Object value, Windows.UI.Xaml.DependencyPropertyValuePrecedences precedence, Windows.UI.Xaml.DependencyPropertyDetails propertyDetails, System.Boolean isPersistentResourceBinding) <0xdde5b90 + 0x00062> in <filename unknown>:0 
        at Windows.UI.Xaml.DependencyObjectStore.SetValue (Windows.UI.Xaml.DependencyProperty property, System.Object value) <0xdde5a48 + 0x00016> in <filename unknown>:0 
        at Windows.UI.Xaml.UIElement.SetValue (Windows.UI.Xaml.DependencyProperty dp, System.Object value) <0xdfa95c0 + 0x0001a> in <filename unknown>:0 
        at Windows.UI.Xaml.Controls.Primitives.Selector.set_SelectedIndex (System.Int32 value) <0xf8c15c8 + 0x00016> in <filename unknown>:0 
        at Windows.UI.Xaml.Controls.FlipView.UpdateSelectedIndex (System.Int32 index) <0xfe8d420 + 0x00024> in <filename unknown>:0 
        at Windows.UI.Xaml.Controls.FlipView.MoveNext () <0xfe8d2a0 + 0x00092> in <filename unknown>:0 
        at Windows.UI.Xaml.Controls.FlipView.OnNextButtonPartClick (System.Object pSender, Windows.UI.Xaml.RoutedEventArgs pArgs) <0xfe8d1f0 + 0x00008> in <filename unknown>:0 
        at Windows.UI.Xaml.Controls.Primitives.ButtonBase.OnClick () <0xfe8c6c0 + 0x00056> in <filename unknown>:0 
        at Windows.UI.Xaml.Controls.Button.OnClick () <0xfe8c540 + 0x0005c> in <filename unknown>:0 
        at Windows.UI.Xaml.Controls.Primitives.ButtonBase.PerformPointerUpAction () <0xfe8c3f8 + 0x0003c> in <filename unknown>:0 
        at Windows.UI.Xaml.Controls.Primitives.ButtonBase.OnPointerReleased (Windows.UI.Xaml.Input.PointerRoutedEventArgs args) <0xfe8c0f8 + 0x00192> in <filename unknown>:0 
        at Windows.UI.Xaml.Controls.Control+<>c.<.cctor>b__388_10 (System.Object sender, Windows.UI.Xaml.Input.PointerRoutedEventArgs args) <0xfe204c8 + 0x00016> in <filename unknown>:0 
        at Windows.UI.Xaml.UIElement.InvokeHandler (System.Object handler, Windows.UI.Xaml.RoutedEventArgs args) <0xf7b2258 + 0x001a0> in <filename unknown>:0 
        at Windows.UI.Xaml.UIElement.RaiseEvent (Windows.UI.Xaml.RoutedEvent routedEvent, Windows.UI.Xaml.RoutedEventArgs args, Windows.UI.Xaml.UIElement+BubblingContext ctx) <0xfaefba0 + 0x00152> in <filename unknown>:0 
        at Windows.UI.Xaml.UIElement.RaisePointerEvent (Windows.UI.Xaml.RoutedEvent evt, Windows.UI.Xaml.Input.PointerRoutedEventArgs args, Windows.UI.Xaml.UIElement+BubblingContext ctx) <0xfadf0c8 + 0x0003e> in <filename unknown>:0 

^ just comment out:
    FlipView::OnSelectedIndexChanged
        bool succeeded = (bool)(m_tpScrollViewer?.CancelDirectManipulations());

## Mouseover cause index reset
> Mousing over the item area cause the selection to reset back to item 0
> You can click `>` next button multiple that works, but it always resets back to 1st

stack trace:
    @xy WindowManagerInterop.ScrollTo({ HtmlId = 275330, horizontalOffset = 0, verticalOffset = , disableAnimation = True })
    at Windows.UI.Xaml.Controls.ScrollContentPresenter.ScrollTo (System.Nullable`1[T] horizontalOffset, System.Nullable`1[T] verticalOffset, System.Boolean disableAnimation) [0x00000] in <filename unknown>:0 
    at Windows.UI.Xaml.Controls.ScrollViewer.ChangeViewNative (System.Nullable`1[T] horizontalOffset, System.Nullable`1[T] verticalOffset, System.Nullable`1[T] zoomFactor, System.Boolean disableAnimation) [0x00000] in <filename unknown>:0 
    at Windows.UI.Xaml.Controls.ScrollViewer.ChangeViewCore (System.Nullable`1[T] horizontalOffset, System.Nullable`1[T] verticalOffset, System.Nullable`1[T] zoomFactor, System.Boolean disableAnimation, System.Boolean shouldSnap) [0x00000] in <filename unknown>:0 
    at Windows.UI.Xaml.Controls.ScrollViewer.ChangeView (System.Nullable`1[T] horizontalOffset, System.Nullable`1[T] verticalOffset, System.Nullable`1[T] zoomFactor, System.Boolean disableAnimation) [0x00000] in <filename unknown>:0 
    at Windows.UI.Xaml.Controls.ScrollViewer.ChangeViewForOrientation (Windows.UI.Xaml.Controls.Orientation orientation, System.Double scrollAdjustment) [0x00000] in <filename unknown>:0 
    at Windows.UI.Xaml.Controls.ScrollViewer.TrimOverscroll (Windows.UI.Xaml.Controls.Orientation orientation) [0x00000] in <filename unknown>:0 
    at Windows.UI.Xaml.Controls.ScrollViewer.ArrangeOverride (Windows.Foundation.Size finalSize) [0x00000] in <filename unknown>:0 
    at Windows.UI.Xaml.FrameworkElement.InnerArrangeCore (Windows.Foundation.Rect finalRect) [0x00000] in <filename unknown>:0 
    at Windows.UI.Xaml.FrameworkElement.ArrangeCore (Windows.Foundation.Rect finalRect) [0x00000] in <filename unknown>:0 
    at Windows.UI.Xaml.UIElement.DoArrange (Windows.Foundation.Rect finalRect) [0x00000] in <filename unknown>:0 
    at Windows.UI.Xaml.UIElement.Arrange (Windows.Foundation.Rect finalRect) [0x00000] in <filename unknown>:0 
    at Windows.UI.Xaml.FrameworkElement.ArrangeElement (Windows.UI.Xaml.UIElement view, Windows.Foundation.Rect finalRect) [0x00000] in <filename unknown>:0 
    at Windows.UI.Xaml.Controls.Grid.InnerArrangeOverride (Windows.Foundation.Size finalSize) [0x00000] in <filename unknown>:0 
    // ...skipping multiple layers of various ArrangeSomething calls
    at Windows.UI.Xaml.FrameworkElement.InnerArrangeCore (Windows.Foundation.Rect finalRect) [0x00000] in <filename unknown>:0 
    at Windows.UI.Xaml.FrameworkElement.ArrangeCore (Windows.Foundation.Rect finalRect) [0x00000] in <filename unknown>:0 
    at Windows.UI.Xaml.UIElement.DoArrange (Windows.Foundation.Rect finalRect) [0x00000] in <filename unknown>:0 
    at Windows.UI.Xaml.UIElement.ArrangeVisualTreeRoot (Windows.Foundation.Rect finalRect) [0x00000] in <filename unknown>:0 
    at Windows.UI.Xaml.UIElement.Arrange (Windows.Foundation.Rect finalRect) [0x00000] in <filename unknown>:0 
    at Windows.UI.Xaml.Window.DispatchInvalidateMeasure () [0x00000] in <filename unknown>:0 
    at Windows.UI.Xaml.Window.<InnerInvalidateMeasure>b__72_0 () [0x00000] in <filename unknown>:0 
    at Uno.UI.Dispatching.CoreDispatcher.InvokeOperation (Uno.UI.Dispatching.UIAsyncOperation operation) [0x0005f] in <filename unknown>:0 
    at Uno.UI.Dispatching.CoreDispatcher.InvokeOperationSafe (Uno.UI.Dispatching.UIAsyncOperation operation) [0x00002] in <filename unknown>:0 
    at Uno.UI.Dispatching.CoreDispatcher.DispatchItems () [0x000ea] in <filename unknown>:0 
    at Uno.UI.Dispatching.CoreDispatcher.DispatcherCallback () [0x00001] in <filename unknown>:0 

there is no pointer events resetting the selection in flipview
    instead it is the scrollviewer attempting to recover from over scrolling
        TrimOverscroll
            ItemsPresenter#275778
            { presenterViewportSize = 200, contentExtent = 200, offset = 200, viewportEnd = 400, overscroll = -200 }

sv hierarchy: ScrollViewer > Border > Grid > ScrollContentPresenter > ItemsPresenter > ItemsStackPanel
    ^ normally, ScrollContentPresenter has a constrained size, with ItemsPresenter & ItemsStackPanel being at full extended size
    ^ in FlipView, they all have the constrained size...

## Scrollability
^ clipping fix might not be needed, if we fix this:
@xy ====== ScrollContentPresenter#278386::MeasureOverride(200,100)
    ^ not measured with infinite width (horizontal)
@xy { 
    CanVerticallyScroll = False,
    CanHorizontallyScroll = False,
    VerticalScrollBarVisibility = Disabled,
    HorizontalScrollBarVisibility = Disabled,
    _forceChangeToCurrentView = False }
    ^ need to set visibility...
```
</details>

tl;dr: 
The ScrollViewer's HV-ScrollBarVisibilities were left at Disabled causing the SV to deem itself not scrollable. After changing selection, on the next arrange pass, the SV::TrimOverscroll will be called to "adjust" the scroll position and in turn triggers FlipView::OnScrollViewerViewChanged that updates the index based on the scroll position.
